synapse/config/workers.py

```diff
@@ -50,13 +52,42 @@ def _instance_to_list_converter(obj: Union[str, List[str]]) -> List[str]:
     return obj
 
 
-@attr.s(auto_attribs=True)
-class InstanceLocationConfig:
+class ConfigModel(BaseModel):
+    """A custom version of Pydantic's BaseModel which
+
+     - ignores unknown fields and
+     - does not allow fields to be overwritten after construction,
+
+    but otherwise uses Pydantic's default behaviour.
+
+    Ignoring unknown fields is a useful default. It means that clients can provide
+    unstable field not known to the server without the request being refused outright.
+
```

Comment:
How about this?

```suggestion
    Ignoring unknown fields is a useful default. There are undocumented settings that exist in Synapse(usually for testing) that would otherwise be refused outright.

```

Reply:
I suggest "For now, ignore unknown fields. In the future, we could change this so that unknown config values cause a validation error, provided the error messages are meaningful to server operators."